### PR TITLE
Feature backup to bot

### PR DESCRIPTION
**Pull request by PopovIV**

### CODE REVIEW

File: src/MangaJet/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -52,7 +55,17 @@
...
+            android:exported="false"
+            android:grantUriPermissions="true">
+            <meta-data
+                android:name="android.support.FILE_PROVIDER_PATHS"
+                android:resource="@xml/filepaths" />
+        </provider>
     </application>
 
-</manifest>
+</manifest>
```
Comment (Sairsey) [inline src/MangaJet/app/src/main/AndroidManifest.xml:71]: Add new line in the end of file

File: src/MangaJet/app/src/main/java/com/mangajet/mangajet/MainActivity.kt [hunk 2]
```diff
@@ -152,21 +157,44 @@ class MainActivity : AppCompatActivity(), ActivityResultCallback<Map<String, Boo
...
         val email = "mangajetmailbot@gmail.com"
+        val tmpFileName = "backupToMail.zip"
 
+        //create zip file
+        var file : File? = null
+        try {
+            // To Download pages, because it won't be in archive
+            if(StorageManager.isExist(tmpFileName, StorageManager.FileType.DownloadedPages))
+                StorageManager.getFile(tmpFileName, StorageManager.FileType.DownloadedPages).delete()
+            if(report.isEmpty()) // Ask me about this later, Vova
```
Comment (Sairsey) [inline src/MangaJet/app/src/main/java/com/mangajet/mangajet/MainActivity.kt:169]: I think I got it.. Some times report is exist but empty. It isn`t a bad thing, you just must check it at start of function.
But Archive you should create only if user wanted to create it. So move archive creation to Dialog "positive" callback

File: src/MangaJet/app/src/main/java/com/mangajet/mangajet/game/GameActivity.kt [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+            val child = group.getChildAt(i)
+            if (child is ImageButton)
+                buttons[i / (emptyX + 1)][i % ((emptyX + 1))] = child
+        }
+        tiles = IntArray(tilesNum + 1)
+        for(i in 0 until group.childCount - 1)
+            tiles[i] = i + 1
+
+        // Picture to 9 parts and to buttons
+        val bitmap = resources.getDrawable(R.drawable.anime_girl).toBitmap()
```
Comment (Sairsey) [inline src/MangaJet/app/src/main/java/com/mangajet/mangajet/game/GameActivity.kt:128]: Maybe we should use some pics from https://konachan.net or https://rule34.paheal.net

File: src/MangaJet/app/src/main/res/layout/activity_game.xml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            android:layout_toEndOf="@id/button8"
+            android:layout_below="@id/button6"
+            android:tag="22"
+            android:layout_width="130dp"
+            android:layout_height="130dp"
+            android:background="@null"
+            ></ImageButton>
+
+    </RelativeLayout>
+</FrameLayout>
```
Comment (Sairsey) [inline src/MangaJet/app/src/main/res/layout/activity_game.xml:96]: Add new line at the end of file

File: src/MangaJet/app/src/main/res/xml/filepaths.xml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<paths>
+    <external-path name="external_files" path="."/>
+</paths>
```
Comment (Sairsey) [inline src/MangaJet/app/src/main/res/xml/filepaths.xml:4]: Add new line in the end of file. And why we need this?

**Review by Sairsey — CHANGES_REQUESTED** [review]

**Review by aVorotnikov — APPROVED** [review]

**Review by Sairsey — APPROVED** [review]

**Pull request merged**